prov/rxm/src/rxm_ep.c

```diff
@@ -885,9 +890,12 @@ rxm_ep_alloc_lmt_tx_res(struct rxm_ep *rxm_ep, struct rxm_conn *rxm_conn,
 	return ret;
 }
 
+/* if the `release_res` is set to 0, caller must ensure release of
+ * the TX buffer and the TX entry */
 static inline ssize_t
 rxm_ep_lmt_tx_send(struct rxm_ep *rxm_ep, struct rxm_conn *rxm_conn,
-		   struct rxm_tx_entry *tx_entry, size_t pkt_size)
+		   struct rxm_tx_entry *tx_entry, size_t pkt_size,
+		   int release_res)
```

Comment:
Adding this parameter is odd.  IMO, the caller that allocates the resource should also handle the release.